src/System.Private.CoreLib/src/System/AppDomain.cs

```diff
@@ -555,107 +437,17 @@ private static object Setup(object arg)
 
                 for (int i = 0; i < propertyNames.Length; i++)
                 {
-                    if (propertyNames[i] == "APPBASE") // make sure in sync with Fusion
-                    {
-                        if (propertyValues[i] == null)
-                            throw new ArgumentNullException("APPBASE");
-
-                        if (PathInternal.IsPartiallyQualified(propertyValues[i]))
-                            throw new ArgumentException(SR.Argument_AbsolutePathRequired);
-
-                        newSetup.ApplicationBase = NormalizePath(propertyValues[i], fullCheck: true);
-                    }
-                    else if (propertyNames[i] == "TRUSTED_PLATFORM_ASSEMBLIES" ||
-                       propertyNames[i] == "PLATFORM_RESOURCE_ROOTS" ||
-                       propertyNames[i] == "APP_PATHS" ||
-                       propertyNames[i] == "APP_NI_PATHS")
-                    {
-                        string values = propertyValues[i];
-                        if (values == null)
-                            throw new ArgumentNullException(propertyNames[i]);
-
-                        ad.SetData(propertyNames[i], NormalizeAppPaths(values));
```

Comment:
The normalization here is unnecessary. The good hosts are going to give us good paths. We do not need to try to normalize the paths here to compensate for broken hosts.

Reply:
Just wondering is maybe we should check that all the paths are absolute. It's relatively subtle that we expect full paths here, and while I agree that we should only work with good hosts, if I'm writing a new host, getting this wrong is easy, but who knows how and where it would fail.

Reply:
I agree that we should validate the arguments, but it does not make sense to parse them here just to do the validation. I have added the validation to where they are actually parsed in unmanaged binder. I know that the diagnostic for errors there is less than ideal ... but it is a step.

Reply:
Sounds good